### DISCUSSION TITLE
redirectTo in generateLink not working

### DISCUSSION THREAD

**Discussion by aunumever** [supabase/supabase#7578] [Questions] [answered]
redirectTo does not seem to be working. I am accessing the app from a particular domain, but the magic link created from the generateLink api call does not support the redirectTo option passed in:

```
			// Get magic link from Supabase
			const { data: userObj, error } = await supabase.auth.api.generateLink('magiclink', email, {
				redirectTo: `${window.location.origin}/app/dashboard`
			});
```

It returns a default `localhost:3000` instead of the redirectUrl: `http://domain.com/app/dashboard`. The domain is set up in my hosts file locally and my app shows different things based on the URL.

**Comment by GaryAustin1** [comment, marked as answer]
Do you have the exact URL set up in SB authentication settings for Additional redirect URLs

**Reply by aunumever** [reply]
Yes, I have added Additional redirect URLs with no success. I have tried adding the port, wildcard subdomains, etc. I have tried 20 different combinations, still only redirects to http://localhost:3000

Additional redirect URLs: 
`http://domain.test:3000,*.domain.test,http://*.domain.test,http://*.domain.test,http://domain.test:3000`

**Reply by GaryAustin1** [reply]
I’ve not used generateLink for magic link just signIn.   But assuming redirectTo works for that, you need the complete url including path like http://domain.com/app/dashboard.

**Reply by aunumever** [reply]
Adding the path after the domain fixed the issue. I had to add `app/dashboard`. I assumed it was just the domain origin that needed to be whitelisted. Thank you.

**Comment by JimmyLv** [comment]
I got it working! the magic is to add the `**` symbol again after each URL 🤣

```shell
https://*.jimmylv.cn/**

https://b.aitodo.co/**
```

**Comment by jide** [comment]
I had the issue while developing locally.
If you do too, the config to change is inside your config.toml file, in additional_redirect_urls.
